File: 0001 - Two Sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> m;
        for(int i = 0; i < nums.size(); i++){
            if(m.count(nums[i])) return {i, m[nums[i]]};
            else m[target-nums[i]] = i;
        }
        return {};
    }
};

File: 0001 - Two Sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        table = {}
        for i in range(len(nums)):
            if nums[i] in table: return [i, table[nums[i]]]
            else : table[target-nums[i]] = i
        return []